packages/components/src/DateField/DateField.tsx

```diff
@@ -0,0 +1,179 @@
+  ThemeExtensionsWithParts,
+  useComponentStyles,
+  useStateProps,
+} from '@marigold/system';
+import { DateSegment } from './DateSegment';
+import { AriaButtonProps } from '@react-aria/button';
+import { FieldBase } from '../FieldBase';
+import { mergeProps } from '@react-aria/utils';
+
+// Props
+// ----------------
+export interface DateFieldThemeExtension
+  extends ThemeExtensionsWithParts<
+    'DateField',
+    ['segment', 'placeholder', 'field', 'calendarButton']
+  > {}
+
+export interface DateFieldProps
+  extends Omit<
+    AriaDateFieldProps<DateValue>,
+    'isDisabled' | 'isReadOnly' | 'isRequired'
+  > {
+  onChange?: (value: DateValue) => void;
+  value?: DateValue;
+  defaultValue?: DateValue;
+  ref?: React.RefObject<unknown> | undefined;
+  iconRight?: ReactElement;
+  iconLeft?: ReactElement;
+  buttonProps?: AriaButtonProps<'button'>;
+  isPressed?: boolean;
+  error?: boolean;
+  errorMessageProps?: HTMLAttributes<HTMLElement>;
+  disabled?: boolean;
+  readOnly?: boolean;
+  required?: boolean;
+  variant?: string;
+  size?: string;
+}
+
+// Component
+// ----------------
+
+export const DateField = ({
+  disabled,
+  readOnly,
+  required,
+  isPressed,
+  error,
+  errorMessage,
+  errorMessageProps,
+  buttonProps,
+  variant,
+  size,
+  iconLeft,
+  iconRight,
+  ...res
+}: DateFieldProps) => {
+  const { locale } = useLocale();
+  const props: AriaDateFieldProps<DateValue> = {
+    isDisabled: disabled,
+    isReadOnly: readOnly,
+    isRequired: required,
+    ...res,
+  };
+  const { label, description } = props;
+  const state = useDateFieldState({
+    ...props,
+    locale,
+    createCalendar,
+  });
+
+  const ref = useRef(null);
+  const { fieldProps, labelProps, descriptionProps } = useDateField(
+    props,
+    state,
+    ref
+  );
+  const styles = useComponentStyles(
+    'DateField',
+    {},
+    { parts: ['segment', 'field', 'calendarButton'] }
+  );
+
+  const stateProps = useStateProps({
+    error,
+    disabled,
+  });
+
+  return (
+    <FieldBase
+      error={error}
+      errorMessage={errorMessage}
+      errorMessageProps={errorMessageProps}
+      label={label}
+      labelProps={{ as: 'span', ...labelProps }}
+      description={description}
+      descriptionProps={descriptionProps}
+      disabled={disabled}
+      stateProps={stateProps}
+      required={required}
+      variant={variant}
+      size={size}
+    >
+      <Box
+        __baseCSS={{
+          display: 'flex',
+          flexDirection: 'row',
+          justifyContent: 'space-between',
+          flexWrap: 'nowrap',
+          borderRadius: '10px',
+          overflow: 'hidden',
+        }}
+        {...mergeProps(fieldProps, stateProps)}
+        css={styles.field}
+      >
+        <Box
+          __baseCSS={{
+            display: 'flex',
+            alignItems: 'center',
+            flexBasis: '100%',
+            gap: '11px',
+            paddingX: '7px',
+          }}
+        >
+          {iconLeft && (
```

Comment:
we have `icon` and `action` for leading and trailing icons. You can have a look on the Autocomplete or Input. Maybe this works too